### DISCUSSION TITLE
Can "Savegame" be transfered to a new installation?

### DISCUSSION THREAD

**Discussion by Kutabu** [ochadenas/cpudefense#129] [Q&A] [answered]
If I want to set up an new phone, can I copy the gamestatus to a new device?
And if yes... how?

**Comment by ochadenas** [comment, marked as answer]
Honestly, I don't know. What's sure is that there is no mechanism provided by the game to export or import game status. I mean, the whole thing started as a little casual game, and I thought that at the most all levels could be played through in one hour or so. I didn't imagine that one day people would have ~ 60 regular levels to play and then still do a hundred more in endless ... 

You can, of course, try to locate the preferences directory, copy the 3 files there, and transfer it to the new device. On my phone it's located at /data/data/de.chadenas.cpudefense/shared_prefs. All game information is stored in these files (not only the preferences, regardless of the name).

I have little experience with the whole Android system. Other games synchronise to the cloud via Google-provided mechanisms, but you need to link the app to a Google account, I think. Which would spoil the advantage that distribution via f-droid just don't require a Google account. Perhaps I will start distributing the app additionally on the Play Store, keeping the f-droid channel as well, but that is more a long-term project.

**Comment by Kutabu** [comment]
Grr... that's bad for me. I use Android 14 and access to this folder is much more difficult. I'd have to root my mobile phone first to get to the files.
Maybe it would be something for the backlog to store the savegames in an accessible way. That would also make sense as an .xml, as it was already considered for the level design. Then players could backup the files on their own. Nothing for now, maybe later, because it isn't that relevant.

Thanks for the support!
